### PR TITLE
chore/LPFG-375

### PR DESCRIPTION
**Pull request by jenbutongit**
**PR includes:** _a lot_

**tl;dr - using svelte API properly**

* All of the components (and page templates) have been updated to use svelte [API](https://svelte.technology). 
  * This means that we are no longer reverse-engineering, wrapping and injecting the svelte compiler.
  * If you wish to import a component, you must import at the bottom of the `*.html` in the `<script>` tag. 

There is a more detailed explanation and some gotchas about how to use it in this project in [lib/src/ui/README.md](lib/src/ui/README.md)

Branched squashed for some clarity. If you care:
Squashed commits: 
[90fb402] minify css
[888a48c] logging and cleaning
[561ed28] check pages, if error, throw
[ea1331e] management ui
[fc3d891] tidy
[10127af] discite course / module / search
[77b8737] split up discite course
[5d63ed4] remove old templating wrapper
[e584389] add svelte script and delete some files
[09b0b2c] search page and update readme
[fbd3969] course pages
[d14478e] profile pages
[3f07458] use $i18n, $datetime, learning record page and suggestions page
[279b582] HomePage and child components
[a1ef446] readme
[a4d198b] start some docs
[d9159f8] WIP
[055248b] using stores
[d2e7f32] wip
[7e0eed4] WIP
[2af2b81] wip
[c3d0094] wip
[fbbbe98] svelte
[be5aab6] refactor discite courses

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,7 +100,7 @@
 		"session-file-store": "1.2.0",
 		"shortid": "2.2.8",
 		"striptags": "3.1.1",
-		"svelte": "1.55.0",
+		"svelte": "^2.6.3",
```
Comment (superafroman) [inline package.json]: specific version

File: src/lib/ui/svelte/shared.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+					output += '</ul>'
+				}
+				output += `<p>${line}</p>`
+			}
+		}
+		return output
+	}
+	return ''
+}
+*/
```
Comment (superafroman) [inline src/lib/ui/svelte/shared.ts]: Can we delete this?

File: src/lib/ui/svelte/shared.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export function allc() {
+	console.log('hello')
+}
```
Comment (superafroman) [inline src/lib/ui/svelte/shared.ts]: Assuming this was a test? Lets get rid of for now unless you see a benefit in having an example?

File: src/lib/ui/template.ts [hunk 4]
```diff
@@ -209,144 +66,73 @@ function toHtml(text: string) {
...
-// ones that need to be regenerated as opposed to wiping the whole cache
-// whenever any template file changes.
-if (!config.PRODUCTION_ENV) {
-	fs.watch(componentDir, {}, resetCache)
-	fs.watch(pageDir, {recursive: true}, resetCache)
+	const store = new Store({
+		...getHelpers(),
+	})
+	res.send(component.render(data, {store}).html)
 }
```
Comment (superafroman) [inline src/lib/ui/template.ts]: This is awesome! I love deleting stuff!!
